crates/inspection-service/src/inspection_service.rs

```diff
@@ -231,3 +231,109 @@ pub fn start_inspection_service(node_config: NodeConfig) {
             .unwrap();
     });
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use futures::executor::block_on;
+    use hyper::body;
+    use once_cell::sync::Lazy;
+    use prometheus::{register_int_counter, IntCounter};
+    use std::io::read_to_string;
+
+    const INT_COUNTER_NAME: &str = "INT_COUNTER";
```

Comment:
nit: Nice! Maybe add a comment saying that this is a test counter only valid for the life of the tests (to ensure the metrics endpoint works).

Reply:
commented